### PR TITLE
lazy data loading for equipments tables

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**
Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

### CODE REVIEW

File: src/components/network/network-explorer.js [hunk 1]
```diff
@@ -105,6 +105,16 @@ const NetworkExplorer = ({
     const [filteredVoltageLevels, setFilteredVoltageLevels] = React.useState(
         []
     );
+    const [substationsLoaded, setSubstationLoaded] = React.useState(false);
```
Comment (FranckLecuyer) [inline src/components/network/network-explorer.js:108]: setSubstationsLoaded for setter name

File: src/components/network/network-map.js [hunk 2]
```diff
@@ -54,6 +55,10 @@ const NetworkMap = forwardRef((props, ref) => {
     const [tooltip, setTooltip] = useState({});
 
     const theme = useTheme();
+
+    const [substationLoaded, setSubstationLoaded] = useState(false);
```
Comment (FranckLecuyer) [inline src/components/network/network-map.js:59]: substationsLoaded for state name and setSubstationsLoaded for setter name

File: src/components/network/nominal-voltage-filter.js [hunk 3]
```diff
@@ -48,6 +48,7 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 const NominalVoltageFilter = (props) => {
+    const [substationsLoaded, setSubstationLoaded] = useState(false);
```
Comment (FranckLecuyer) [inline src/components/network/nominal-voltage-filter.js:51]: setSubstationsLoaded for setter name

File: src/components/network/overloaded-lines-view.js [hunk 4]
```diff
@@ -51,6 +51,7 @@ const useStyles = makeStyles((theme) => ({
 
 const OverloadedLinesView = (props) => {
     const [lines, setLines] = useState(null);
+    const [lineLoaded, setLineLoaded] = useState(false);
```
Comment (FranckLecuyer) [inline src/components/network/overloaded-lines-view.js:54]: linesLoaded for state name and setLinesLoaded for setter name

File: src/components/study-pane.js [hunk 5]
```diff
@@ -173,6 +173,7 @@ const StudyPane = (props) => {
     const lineFlowMode = useSelector((state) => state.lineFlowMode);
 
     const lineFlowColorMode = useSelector((state) => state.lineFlowColorMode);
+    const [substationsLoaded, setSubstationLoaded] = useState(false);
```
Comment (FranckLecuyer) [inline src/components/study-pane.js:177]: setSubstationsLoaded for setter name

File: src/components/study-pane.js [hunk 6]
```diff
@@ -354,87 +355,44 @@ const StudyPane = (props) => {
...
-                network.setBatteries(values[6]);
-                network.setDanglingLines(values[7]);
-                network.setHvdcLines(values[8]);
-                network.setLccConverterStations(values[9]);
-                network.setVscConverterStations(values[10]);
-                network.setShuntCompensators(values[11]);
-                network.setStaticVarCompensators(values[12]);
-
+    const loadNetwork = useCallback(
+        (isJusteUpdate) => {
```
Comment (FranckLecuyer) [inline src/components/study-pane.js:356]: isJustUpdate or isUpdate for argument name

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Maybe the security analysis result could be also loaded in a lazy manner, instead of loading
them during study loading

File: src/components/network/network.js [hunk 7]
```diff
@@ -10,32 +10,77 @@ const elementIdIndexer = (map, element) => {
...
+            if (cbUpdateDone) this.cbUpdateDone.add(cbUpdateDone);
+            if (!this.updating) {
+                this.updating = true;
+                Promise.all([this.fetcher()])
+                    .then((val) => {
+                        this.values = val[0];
+                        if (this.postUpdate) this.postUpdate(this.values);
+                        this.updating = false;
+                        this.cbUpdateDone.forEach((cb) => cb());
+                        this.cbUpdateDone.clear();
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/network/network.js]: when the ressource is already there, we never pass through the Promise, but we still need to set the cbstate

File: src/components/network/overloaded-lines-view.js (hunk 4 above)
Comment (JacquesBorsenberger) [reply src/components/network/overloaded-lines-view.js]: done&done

File: src/components/network/network-explorer.js (hunk 1 above)
Comment (JacquesBorsenberger) [reply src/components/network/network-explorer.js]: done

File: src/components/network/network-map.js (hunk 2 above)
Comment (JacquesBorsenberger) [reply src/components/network/network-map.js]: done

File: src/components/network/nominal-voltage-filter.js (hunk 3 above)
Comment (JacquesBorsenberger) [reply src/components/network/nominal-voltage-filter.js]: done

File: src/components/study-pane.js (hunk 6 above)
Comment (JacquesBorsenberger) [reply src/components/study-pane.js]: done

File: src/components/study-pane.js (hunk 5 above)
Comment (JacquesBorsenberger) [reply src/components/study-pane.js]: done

File: src/components/network/network.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/network/network.js]: I understand, but regardless if the resource is already here or not, cbUpdateDone is called at the end of the method, so why calling it also in the 'then' method of the promise

[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/network/network.js]: when the resource is not there and the component is updating, we return undefined

**Review by FranckLecuyer — APPROVED** [review]

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
No more lines flows update on network map after line switch modification and load flow running

**Comment by jonenst** [comment]
is this superseded by #293  ?
Did you take https://github.com/gridsuite/gridstudy-app/pull/286#pullrequestreview-572050662 into account ?

**Comment by JacquesBorsenberger** [comment]
> is this superseded by #293 ?
> Did you take [#286 (review)](https://github.com/gridsuite/gridstudy-app/pull/286#pullrequestreview-572050662) into account ?

yes

**Pull request closed without merging**